### ISSUE TITLE
[SEVERE] Exception ticking world Overworld/0

### ISSUE THREAD

**Issue by mainsheep** [MinimallyCorrect/TickThreading#418]
I've been getting these kind of errors lately:

2013-04-30 22:56:26 [SEVERE] Exception ticking world Overworld/0 java.util.ConcurrentModificationException         at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:819)         at java.util.ArrayList$Itr.next(ArrayList.java:791)         at codechicken.chunkloader.ChunkLoaderManager.cleanChunks(ChunkLoaderManager.java:821)         at codechicken.chunkloader.ChunkLoaderManager.tickEnd(ChunkLoaderManager.java:853)         at codechicken.chunkloader.ChunkLoaderEventHandler.tickEnd(ChunkLoaderEventHandler.java:39)         at cpw.mods.fml.common.SingleIntervalHandler.tickEnd(SingleIntervalHandler.java:22)         at cpw.mods.fml.common.FMLCommonHandler.tickEnd(FMLCommonHandler.java:79)         at cpw.mods.fml.common.FMLCommonHandler.onPostWorldTick(FMLCommonHandler.java:283)         at net.minecraft.server.MinecraftServer.doWorldTick(MinecraftServer.java:461)         at me.nallar.patched.server.PatchMinecraftServer$TickRunnable.run(PatchMinecraftServer.java:572)         at me.nallar.tickthreading.minecraft.ThreadManager$1.run(ThreadManager.java:41)         at me.nallar.tickthreading.minecraft.ThreadManager$ServerWorkThread.run(ThreadManager.java:157)

The above error is from Overworld, but the error also come with End, TLF and Nether.

I'm running v1473 TT, and L21 MCPC-Legacy. All running Mindcrack 8.2.0 from Ubuntu Server 12.04 and Java 7.

**Comment by LunNova** [comment]
That method should be disabled by the patcher, patcher log please?

**Comment by mainsheep** [comment]
Where can I find the patcher log?

**Comment by mainsheep** [comment]
Found it.

http://pastebin.com/x3G5E27L

**Comment by mainsheep** [comment]
I'm still getting these errors, now with TT 1488 and MCPC+ L22.

Console : 2013-05-02 09:27:46 [SEVERE] Exception ticking world Twilight Forest/7
Console : java.util.ConcurrentModificationException
Console :       at java.util.HashMap$HashIterator.nextEntry(HashMap.java:839)
Console :       at java.util.HashMap$ValueIterator.next(HashMap.java:868)
Console :       at codechicken.chunkloader.ChunkLoaderManager.revivePlayerLoaders(ChunkLoaderManager.java:910)
Console :       at codechicken.chunkloader.ChunkLoaderManager.tickEnd(ChunkLoaderManager.java:856)
Console :       at codechicken.chunkloader.ChunkLoaderEventHandler.tickEnd(ChunkLoaderEventHandler.java:39)
Console :       at cpw.mods.fml.common.SingleIntervalHandler.tickEnd(SingleIntervalHandler.java:22)
Console :       at cpw.mods.fml.common.FMLCommonHandler.tickEnd(FMLCommonHandler.java:79)
Console :       at cpw.mods.fml.common.FMLCommonHandler.onPostWorldTick(FMLCommonHandler.java:283)
Console :       at net.minecraft.server.MinecraftServer.doWorldTick(MinecraftServer.java:462)
Console :       at net.minecraft.server.MinecraftServer.r(MinecraftServer.java:349)
Console :       at ho.r(DedicatedServer.java:309)
Console :       at net.minecraft.server.MinecraftServer.q(MinecraftServer.java:268)
Console :       at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:181)
Console :       at fy.run(ThreadMinecraftServer.java:16)
Console : 2013-05-02 09:27:46 [SEVERE] Exception ticking world Nether/-1
Console : java.util.ConcurrentModificationException
Console :       at java.util.HashMap$HashIterator.nextEntry(HashMap.java:839)
Console :       at java.util.HashMap$ValueIterator.next(HashMap.java:868)
Console :       at codechicken.chunkloader.ChunkLoaderManager.revivePlayerLoaders(ChunkLoaderManager.java:910)
Console :       at codechicken.chunkloader.ChunkLoaderManager.tickEnd(ChunkLoaderManager.java:856)
Console :       at codechicken.chunkloader.ChunkLoaderEventHandler.tickEnd(ChunkLoaderEventHandler.java:39)
Console :       at cpw.mods.fml.common.SingleIntervalHandler.tickEnd(SingleIntervalHandler.java:22)
Console :       at cpw.mods.fml.common.FMLCommonHandler.tickEnd(FMLCommonHandler.java:79)
Console :       at cpw.mods.fml.common.FMLCommonHandler.onPostWorldTick(FMLCommonHandler.java:283)
Console :       at net.minecraft.server.MinecraftServer.doWorldTick(MinecraftServer.java:462)
Console :       at net.minecraft.server.MinecraftServer.r(MinecraftServer.java:349)
Console :       at ho.r(DedicatedServer.java:309)
Console :       at net.minecraft.server.MinecraftServer.q(MinecraftServer.java:268)
Console :       at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:181)
Console :       at fy.run(ThreadMinecraftServer.java:16)
Console : 2013-05-02 09:27:46 [SEVERE] Exception ticking world The End/1
Console : java.util.ConcurrentModificationException
Console :       at java.util.HashMap$HashIterator.nextEntry(HashMap.java:839)
Console :       at java.util.HashMap$ValueIterator.next(HashMap.java:868)
Console :       at codechicken.chunkloader.ChunkLoaderManager.revivePlayerLoaders(ChunkLoaderManager.java:910)
Console :       at codechicken.chunkloader.ChunkLoaderManager.tickEnd(ChunkLoaderManager.java:856)
Console :       at codechicken.chunkloader.ChunkLoaderEventHandler.tickEnd(ChunkLoaderEventHandler.java:39)
Console :       at cpw.mods.fml.common.SingleIntervalHandler.tickEnd(SingleIntervalHandler.java:22)
Console :       at cpw.mods.fml.common.FMLCommonHandler.tickEnd(FMLCommonHandler.java:79)
Console :       at cpw.mods.fml.common.FMLCommonHandler.onPostWorldTick(FMLCommonHandler.java:283)
Console :       at net.minecraft.server.MinecraftServer.doWorldTick(MinecraftServer.java:462)
Console :       at net.minecraft.server.MinecraftServer.r(MinecraftServer.java:349)
Console :       at ho.r(DedicatedServer.java:309)
Console :       at net.minecraft.server.MinecraftServer.q(MinecraftServer.java:268)
Console :       at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:181)
Console :       at fy.run(ThreadMinecraftServer.java:16)

The latest patcher log when patching newest TT and MCPC+ build:
http://pastebin.com/WgXu6qrm

**Comment by LunNova** [comment]
Can you give the console log, up to the "server started on IP address/port" bit?

**Comment by mainsheep** [comment]
I think the problem came with installing different version of TT.
I installed the newest TT which came out today, and it stated "this map was saved with 1473, and might not work". So I rolled back to that version, and the problem is gone. Lately we have had some freezes and crashes, around 10-15 minutes in between, which seem to have disapeared when installing the older version.

In any case, here is the console output from when I had a version of TT from yesterday installed with MCPC+ L22 installed: http://pastebin.com/xGCAC7Wk
